src/data/java_profile.rs

```diff
@@ -0,0 +1,340 @@
+            Ok(jps_out) => {
+                jps_str = String::from_utf8(jps_out.stdout).unwrap_or_default();
+                let jps: Vec<&str> = jps_str.split_whitespace().collect();
+                for i in (0..jps.len()).step_by(2) {
+                    if jps[i + 1] != "Jps" {
+                        self.process_map
+                            .insert(String::from(jps[i]), vec![String::from(jps[i + 1])]);
+                    }
+                }
+            }
+            Err(e) => {
+                error!("Jps command failed: {}", e);
+            }
+        }
+        jps_str
+    }
+
+    fn launch_pgrep(&mut self) -> Vec<String> {
+        let pgrep_cmd = Command::new("pgrep").arg("java").output();
+        let mut pgrep: Vec<String> = Vec::new();
+        match pgrep_cmd {
+            Ok(pgrep_out) => {
+                let pgrep_str = String::from_utf8(pgrep_out.stdout).unwrap();
+                pgrep = pgrep_str
+                    .split_whitespace()
+                    .map(|s| s.to_string())
+                    .collect();
+            }
+            Err(e) => {
+                error!("Pgrep command failed: {}", e);
+            }
+        }
+        pgrep
+    }
+}
+
+impl CollectData for JavaProfileRaw {
+    fn prepare_data_collector(&mut self, params: CollectorParams) -> Result<()> {
+        let mut jids: Vec<String> = Vec::new();
+        let pgrep: Vec<String> = self.launch_pgrep();
+        for pid in pgrep {
+            if self.process_map.contains_key(&pid) {
+                continue;
+            }
+            self.process_map
+                .insert(pid.clone(), vec![String::from("Could not resolve name!")]);
+        }
+
+        let jps_str = self.update_process_map();
+        let jps: Vec<&str> = jps_str.split_whitespace().collect();
+
+        let default = String::from("jps");
+        let jprofile = params
+            .profile
+            .get(JAVA_PROFILE_FILE_NAME)
+            .unwrap_or(&default)
+            .as_str();
+        match jprofile {
+            "jps" => {
+                jids = self.process_map.clone().into_keys().collect();
+                debug!("Jps will be run if new JVM is started during aperf record to resolve process names.",);
+            }
+            _ => {
+                let args: Vec<&str> = jprofile.split(',').collect();
+                for arg in args {
+                    if !jps.contains(&arg) {
+                        error!("No JVM with name/PID '{}'.", arg);
+                        continue;
+                    } else if jps.iter().position(|&r| r == arg).unwrap()
+                        != jps.iter().rposition(|&r| r == arg).unwrap()
+                    {
+                        error!("Multiple JVMs with the name '{}', please provide PID.", arg);
+                        continue;
+                    }
+                    if let Some(jid) = self.get_jid(arg) {
+                        jids.push(jid);
+                    }
```

Comment:
How would it be possible this is None, given the if statement a couple lines up? We need to be careful about the various data structures we have and how they can be inconsistent with respect to eachother.

And if this legitimately can happen, it should be reflected to the user somehow, not silently dropped.

Reply:
You're right, it is not possible to be none. Changed to handle inside of `get_jid` function.